google/auth/pluggable.py

```diff
@@ -0,0 +1,244 @@
+            client_id=client_id,
+            client_secret=client_secret,
+            quota_project_id=quota_project_id,
+            scopes=scopes,
+            default_scopes=default_scopes,
+            workforce_pool_user_project=workforce_pool_user_project,
+        )
+        if not isinstance(credential_source, Mapping):
+            self._credential_source_executable = None
+            raise ValueError(
+                "Missing credential_source. The credential_source is not a dict."
+            )
+        else:
+            self._credential_source_executable = credential_source.get("executable")
+            if not self._credential_source_executable:
+                raise ValueError(
+                    "Missing credential_source. An 'executable' must be provided."
+                )
+            self._credential_source_executable_command = self._credential_source_executable.get("command")
+            self._credential_source_executable_timeout_millis = self._credential_source_executable.get("timeout_millis")
+            self._credential_source_executable_output_file = self._credential_source_executable.get("output_file")
+
+            # environment_id is only supported in AWS or dedicated future external
+            # account credentials.
+            if "environment_id" in credential_source:
+                raise ValueError(
+                    "Invalid Pluggable credential_source field 'environment_id'"
+                )
+
+        if not self._credential_source_executable_command:
+            raise ValueError(
+                "Missing command. Executable command must be provided."
+            )
+        if not self._credential_source_executable_timeout_millis:
+            raise ValueError(
+                "Missing timeout_millis. Executable timeout millis must be provided."
+            )
+
+    @_helpers.copy_docstring(external_account.Credentials)
+    def retrieve_subject_token(self, request):
+        env_allow_executables = os.environ.get('GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES')
+        if env_allow_executables is None or env_allow_executables != '1':
+            raise ValueError(
+                "Executables need to be explicitly allowed to run."
+            )
+        
+        # Inject env vars
+        os.environ["GOOGLE_EXTERNAL_ACCOUNT_AUDIENCE"] = self._audience
+        os.environ["GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE"] = self._subject_token_type
+        os.environ["GOOGLE_EXTERNAL_ACCOUNT_INTERACTIVE"] = "0" # Always set to 0 until interactive mode is implemented.
+        if self._service_account_impersonation_url is not None:
+            os.environ["GOOGLE_EXTERNAL_ACCOUNT_IMPERSONATED_EMAIL"] = self._service_account_impersonation_url
+        if self._credential_source_executable_output_file is not None:
+            os.environ["GOOGLE_EXTERNAL_ACCOUNT_OUTPUT_FILE"] = self._credential_source_executable_output_file
```

Comment:
do we need to reset these env vars back to their original values after calling `subprocess.run`?

Reply:
Hmm, good point. I don't see a clear issue without resetting it for now. But to be safe, I think it's better to reset them back.